packages/engine/Source/Core/Resource.js

```diff
@@ -555,6 +388,138 @@ Resource.prototype.toString = function () {
   return this.getUrlComponent(true, true);
 };
 
+/**
+ * Parse a url string, and store its info
+ *
+ * @param {string} url The input url string.
+ * @param {boolean} merge If true, we'll merge with the resource's existing queryParameters. Otherwise they will be replaced.
+ * @param {boolean} preserveQuery If true duplicate parameters will be concatenated into an array. If false, keys in url will take precedence.
+ * @param {string} [baseUrl] If supplied, and input url is a relative url, it will be made absolute relative to baseUrl
+ *
+ * @private
+ */
+Resource.prototype.parseUrl = function (url, merge, preserveQuery, baseUrl) {
+  let uri = new Uri(url);
+  const query = parseQueryString(uri.query());
+
+  this._queryParameters = merge
+    ? combineQueryParameters(query, this.queryParameters, preserveQuery)
+    : query;
+
+  // Remove unneeded info from the Uri
+  uri.search("");
+  uri.fragment("");
+
+  if (defined(baseUrl) && uri.scheme() === "") {
+    uri = uri.absoluteTo(getAbsoluteUri(baseUrl));
+  }
+
+  this._url = uri.toString();
+};
+
+/**
+ * Parses a query string and returns the object equivalent.
+ *
+ * @param {string} queryString The query string
+ * @returns {object}
+ *
+ * @private
+ */
+function parseQueryString(queryString) {
+  if (queryString.length === 0) {
+    return {};
```

Comment:
The previous iteration of this function did not create new objects. Could we pass in a result parameter?

Reply:
This would require a change to `Core/queryToObject`, which currently creates a new result object. I could do that here, but I was thinking of dropping that function entirely when we resolve https://github.com/CesiumGS/cesium/issues/11168.